core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -1838,11 +1832,7 @@ private static List<HistoryEntry> updateSnapshotLog(
       Set<Long> intermediateSnapshotIds = intermediateSnapshotIdSet(changes, currentSnapshotId);
       boolean hasIntermediateSnapshots = !intermediateSnapshotIds.isEmpty();
       boolean hasRemovedSnapshots =
-          changes.stream()
-              .anyMatch(
-                  change ->
-                      change instanceof MetadataUpdate.RemoveSnapshots
-                          || change instanceof MetadataUpdate.RemoveSnapshot);
```

Comment:
Shouldn't this part remain as is? That way if a server is handling the bulk update we still update the history entries

Reply:
Sounds good. Let me revert this minimize the change.